yt/funcs.py

```diff
@@ -1310,3 +1310,13 @@ def sglob(pattern):
     Return the results of a glob through the sorted() function.
     """
     return sorted(glob.glob(pattern))
+
+
+class DictWithFactory(dict):
```

Comment:
I'm not convinced a _class_ belongs in `funcs.py` (even though there are already a handful of these living here)